codec-http2/src/main/java/io/netty/handler/codec/http2/StreamBufferingEncoder.java

```diff
@@ -89,7 +88,7 @@ public long errorCode() {
             return errorCode;
         }
 
-        public ByteString debugData() {
+        public byte[] debugData() {
```

Comment:
> Do we need to call `close()` on the array ?

Do you mean `clone`?  I don't think that is necessary as it is already a copy from the original data.

> should me maybe return a ByteBuf or ByteBuffer ?

@nmittler and I discussed this above https://github.com/netty/netty/pull/4413#discussion_r43528906.  The issue with `ByteBuf` is the reference counting. The exception is used to complete a promise, which may not be expecting to have to release any `ByteBuf` payload in the `Exception`.  I guess we could use `nioBufferCount` and `nioBuffer` to see if we can get a `ByteBuffer` ... but if it doesn't we still may have to copy anyways.  Also it is a bit awkward exposing a `ByteBuffer` to the user. @nmittler - WDYT?